doc/chapters/configuration/index.rst

```diff
@@ -49,4 +49,33 @@ As it is indicated in :ref:`design chapter <design>`, each IPC is managed by a g
 
 CMake Options
 =============
-CMake options can be used to configure softwareContainer. Desired gateways can be enabled/disabled by passing flags to cmake. (Currently following gateways are able for enable/disable option; network, pulse, device node, dbus, cgroup :ref:`Gateways <gateways>`).
+
+CMake options can be used to configure softwareContainer. The CMake options can be categorized into four groups:
+
+1. Enabling / disabling gateways.
+2. Enabling / disabling generation of documentation.
+3. Network
+4. Testing
+
+Gateways options
+----------------
+If a gateway is disabled this means that the gateway will not be apart of SoftwareContainer at compile time. Hence this will disable the IPC mechanism that the gateway in question handles.
+As an example, the following snippet will install SoftwareContainer without any network support::
+
+    cmake -DENABLE_NETWORKGATEWAY=OFF ..
+    make
+    sudo make install
+
+A complete list of available gateways can be found at :ref:`Gateways <gateways>`.
+
+Documentation options
+---------------------
+These options allows specification of what documentation to generate when running make.
+
+Network
+-------
+Options governing how SoftwareContainer should react when establishing network access.
+
+Testing
+-------
+Options used when running tests, such as timing and coverage requirements.
```

Comment:
Would be nice with a list of the options even if they aren't properly documented on effects for now.